### DISCUSSION TITLE
UI index page suggest that “Please set a root dashboard for GENERAL”

### DISCUSSION THREAD

**Discussion by miracle-yyh** [apache/skywalking#8951] [Q&A] [answered]
I pulled the code for Skywalking 9.0, built the OAP and UI mirrors locally, and started them. Localhost: 8080 with a browser, the page appears normal, but something is missing. The page looks like this：
![image](https://user-images.githubusercontent.com/52189503/165228279-f30aa22a-9751-4dcb-b73d-703cc1efa64c.png)

This problem has been bothering me all day, what should I do and how to  set a root dashboard for GENERAL？

**Comment by wu-sheng** [comment, marked as answer]
All roots have been set in release. This means OAP is not working. Make sure you are using clean Database env.
Demo is using default dashboards, check showcase doc.

**Reply by miracle-yyh** [reply]
Thank you very much！
To some extent，it's true that OAP is not working: UI is failed to connect to the OAP.The real reson is that the SW_OAP_ADDRESS i 
configured in starting command shouldn't be “http://127.0.0.1:12800” but the container ip.
And the oap-server/server-starter/src/main/resources/application.yml file seems to be missing the driverClassName configuration of mysql，after adding property and dependency，setting mysql as storage is working.

**Comment by zapvol** [comment]
I have the same problem，I have read the upgrade documentation [v9-version-upgrade](https://skywalking.apache.org/docs/main/v9.0.0/en/faq/v9-version-upgrade/)，
ES delete ui_template,sw_service_traffic_xxxx,sw_instance_traffic_xxx

OAP is working 
![image](https://user-images.githubusercontent.com/18160974/165363728-bd4db7fd-f392-4cbb-8524-3c24f86e3bed.png)

ui is working
env 
```
SW_OAP_ADDRESS="http://sky-oap:12800"
SW_TIMEOUT="20000"
```
![image](https://user-images.githubusercontent.com/18160974/165364052-2ff30d5e-4f04-458d-911b-53ea738412df.png)

but ui page has not showing properly.

**Reply by wu-sheng** [reply]
The easiest way is find a k8s, nd run showcase to demo, which is demo.skywalking.apache.org shows.

**Reply by zapvol** [reply]
> The easiest way is find a k8s, nd run showcase to demo, which is demo.skywalking.apache.org shows.

I don't have k8s, just run it locally. I have cleared all ES data and restart, still the same result.

```
version: '3'

services:
  sky-oap:
    container_name: sky-oap
    image: sky-oap-server:8-jre-slim
    build:
      context: ./skywalking/oap-server
    env_file:
      - ./env/sky-oap.env
    restart: always
    ports:
      - "11800:11800"
    networks:
      - database
      - public

  sky-ui:
    container_name: sky-ui
    image: sky-ui:8-jre-slim
    build:
      context: ./skywalking/ui
    env_file:
      - ./env/sky-ui.env
    depends_on:
      - sky-oap
    restart: always
    ports:
      - "8081:8080"
    networks:
      - public
 ```

archive dockerfile is
```
FROM alpine:3.15 AS build
ARG version=9.0.0
RUN echo "Building Apache SkyWalking ${version}"
# downloaded locally
COPY apache-skywalking-apm-9.0.0.tar.gz /tmp/apache-skywalking-apm.tar.gz
COPY apache-skywalking-java-agent-8.10.0.tgz /tmp/apache-skywalking-java-agent.tgz
COPY skywalking-cli-0.10.0-bin.tgz /tmp/skywalking-cli-bin.tgz

RUN set -eux; \
    cd /tmp; \
    mkdir -p /skywalking/apm; \
    tar -zxf apache-skywalking-apm.tar.gz --strip-components=1 -C /skywalking/apm; \
    rm -rf /skywalking/apm/bin/; \
    mkdir -p /skywalking/java-agent; \
    tar -zxf apache-skywalking-java-agent.tgz -C /skywalking/java-agent; \
    mkdir skywalking-cli-bin; \
    tar -zxf skywalking-cli-bin.tgz --strip-components=1 -C skywalking-cli-bin; \
    mkdir -p /skywalking/cli-bin; \
    mv skywalking-cli-bin/bin/swctl-0.10.0-linux-amd64 /skywalking/cli-bin/swctl; \
    chmod 755 /skywalking/cli-bin/swctl; \
    ls -la /skywalking; \
    rm -rf /tmp/*

FROM alpine:3.15

LABEL maintainer="jonathan <jianyang1209@outlook.com>"

COPY --from=build /skywalking /skywalking

WORKDIR /skywalking
```
will build as alpine:skywalking

my oap dockerfile is
```
FROM alpine:skywalking as skywalking

WORKDIR /skywalking/apm

RUN set -eux; \
    rm -rf "config/log4j2.xml"; \
    rm -rf "webapp";

COPY log4j2.xml ./config/log4j2.xml
COPY docker-entrypoint.sh ./docker-entrypoint.sh

RUN set -eux; \
    chmod +x docker-entrypoint.sh;

FROM openjdk:8-jre-slim

LABEL maintainer="jonathan <jianyang1209@outlook.com>"

COPY --from=skywalking /skywalking/apm /skywalking

WORKDIR /skywalking

EXPOSE 12800 11800 1234

ENTRYPOINT ["./docker-entrypoint.sh"]
```
docker-entrypoint.sh is
```
#!/bin/bash

set -ex
JAVA_OPTS="${JAVA_OPTS:-  -Xms256M -Xmx512M}"
OAP_HOME=/skywalking

OAP_LOG_DIR="${OAP_HOME}/logs"
if [ ! -d "${OAP_LOG_DIR}" ]; then
    mkdir -p "${OAP_LOG_DIR}"
fi

CLASSPATH="$OAP_HOME/config:$CLASSPATH"
for i in "$OAP_HOME"/oap-libs/*.jar
do
    CLASSPATH="$i:$CLASSPATH"
done

exec java ${JAVA_OPTS} -classpath ${CLASSPATH} org.apache.skywalking.oap.server.starter.OAPServerStartUp
```
sky-oap.env is
```
TZ=Asia/Shanghai
JAVA_OPTS=" -Xms256M -Xmx512M "
SW_STORAGE=elasticsearch
SW_STORAGE_ES_CLUSTER_NODES=elasticsearch:9200
```

ui
docker-entrypoint.sh
```
#!/bin/bash

set -ex
WEBAPP_HOME=/skywalking

WEBAPP_LOG_DIR="${WEBAPP_LOG_DIR:-${WEBAPP_HOME}/logs}"
JAVA_OPTS="${JAVA_OPTS:-  -Xms256M -Xmx512M}"
JAR_PATH="${WEBAPP_HOME}/webapp"

if [ ! -d "${WEBAPP_LOG_DIR}" ]; then
    mkdir -p "${WEBAPP_LOG_DIR}"
fi
LOG_FILE_LOCATION=${WEBAPP_LOG_DIR}/webapp.log

exec java ${JAVA_OPTS} -jar ${JAR_PATH}/skywalking-webapp.jar \
         --spring.config.location=${JAR_PATH}/webapp.yml \
         --logging.file=${LOG_FILE_LOCATION}
```

sky-ui.env
```
TZ=Asia/Shanghai
JAVA_OPTS=" -Xms256M -Xmx512M "
SW_OAP_ADDRESS="http://sky-oap:12800"
SW_TIMEOUT=20000
```

**Reply by miracle-yyh** [reply]
> > The easiest way is find a k8s, nd run showcase to demo, which is demo.skywalking.apache.org shows.
> 
> I don't have k8s, just run it locally. I have cleared all ES data and restart, still the same result.
> 
> ```
> version: '3'
> 
> services:
>   sky-oap:
>     container_name: sky-oap
>     image: sky-oap-server:8-jre-slim
>     build:
>       context: ./skywalking/oap-server
>     env_file:
>       - ./env/sky-oap.env
>     restart: always
>     ports:
>       - "11800:11800"
>     networks:
>       - database
>       - public
> 
>   sky-ui:
>     container_name: sky-ui
>     image: sky-ui:8-jre-slim
>     build:
>       context: ./skywalking/ui
>     env_file:
>       - ./env/sky-ui.env
>     depends_on:
>       - sky-oap
>     restart: always
>     ports:
>       - "8081:8080"
>     networks:
>       - public
> ```
> 
> archive dockerfile is
> 
> ```
> FROM alpine:3.15 AS build
> ARG version=9.0.0
> RUN echo "Building Apache SkyWalking ${version}"
> # downloaded locally
> COPY apache-skywalking-apm-9.0.0.tar.gz /tmp/apache-skywalking-apm.tar.gz
> COPY apache-skywalking-java-agent-8.10.0.tgz /tmp/apache-skywalking-java-agent.tgz
> COPY skywalking-cli-0.10.0-bin.tgz /tmp/skywalking-cli-bin.tgz
> 
> RUN set -eux; \
>     cd /tmp; \
>     mkdir -p /skywalking/apm; \
>     tar -zxf apache-skywalking-apm.tar.gz --strip-components=1 -C /skywalking/apm; \
>     rm -rf /skywalking/apm/bin/; \
>     mkdir -p /skywalking/java-agent; \
>     tar -zxf apache-skywalking-java-agent.tgz -C /skywalking/java-agent; \
>     mkdir skywalking-cli-bin; \
>     tar -zxf skywalking-cli-bin.tgz --strip-components=1 -C skywalking-cli-bin; \
>     mkdir -p /skywalking/cli-bin; \
>     mv skywalking-cli-bin/bin/swctl-0.10.0-linux-amd64 /skywalking/cli-bin/swctl; \
>     chmod 755 /skywalking/cli-bin/swctl; \
>     ls -la /skywalking; \
>     rm -rf /tmp/*
> 
> FROM alpine:3.15
> 
> LABEL maintainer="jonathan <jianyang1209@outlook.com>"
> 
> COPY --from=build /skywalking /skywalking
> 
> WORKDIR /skywalking
> ```
> 
> will build as alpine:skywalking
> 
> my oap dockerfile is
> 
> ```
> FROM alpine:skywalking as skywalking
> 
> WORKDIR /skywalking/apm
> 
> RUN set -eux; \
>     rm -rf "config/log4j2.xml"; \
>     rm -rf "webapp";
> 
> COPY log4j2.xml ./config/log4j2.xml
> COPY docker-entrypoint.sh ./docker-entrypoint.sh
> 
> RUN set -eux; \
>     chmod +x docker-entrypoint.sh;
> 
> FROM openjdk:8-jre-slim
> 
> LABEL maintainer="jonathan <jianyang1209@outlook.com>"
> 
> COPY --from=skywalking /skywalking/apm /skywalking
> 
> WORKDIR /skywalking
> 
> EXPOSE 12800 11800 1234
> 
> ENTRYPOINT ["./docker-entrypoint.sh"]
> ```
> 
> docker-entrypoint.sh is
> 
> ```
> #!/bin/bash
> 
> set -ex
> JAVA_OPTS="${JAVA_OPTS:-  -Xms256M -Xmx512M}"
> OAP_HOME=/skywalking
> 
> OAP_LOG_DIR="${OAP_HOME}/logs"
> if [ ! -d "${OAP_LOG_DIR}" ]; then
>     mkdir -p "${OAP_LOG_DIR}"
> fi
> 
> CLASSPATH="$OAP_HOME/config:$CLASSPATH"
> for i in "$OAP_HOME"/oap-libs/*.jar
> do
>     CLASSPATH="$i:$CLASSPATH"
> done
> 
> exec java ${JAVA_OPTS} -classpath ${CLASSPATH} org.apache.skywalking.oap.server.starter.OAPServerStartUp
> ```
> 
> sky-oap.env is
> 
> ```
> TZ=Asia/Shanghai
> JAVA_OPTS=" -Xms256M -Xmx512M "
> SW_STORAGE=elasticsearch
> SW_STORAGE_ES_CLUSTER_NODES=elasticsearch:9200
> ```
> 
> ui docker-entrypoint.sh
> 
> ```
> #!/bin/bash
> 
> set -ex
> WEBAPP_HOME=/skywalking
> 
> WEBAPP_LOG_DIR="${WEBAPP_LOG_DIR:-${WEBAPP_HOME}/logs}"
> JAVA_OPTS="${JAVA_OPTS:-  -Xms256M -Xmx512M}"
> JAR_PATH="${WEBAPP_HOME}/webapp"
> 
> if [ ! -d "${WEBAPP_LOG_DIR}" ]; then
>     mkdir -p "${WEBAPP_LOG_DIR}"
> fi
> LOG_FILE_LOCATION=${WEBAPP_LOG_DIR}/webapp.log
> 
> exec java ${JAVA_OPTS} -jar ${JAR_PATH}/skywalking-webapp.jar \
>          --spring.config.location=${JAR_PATH}/webapp.yml \
>          --logging.file=${LOG_FILE_LOCATION}
> ```
> 
> sky-ui.env
> 
> ```
> TZ=Asia/Shanghai
> JAVA_OPTS=" -Xms256M -Xmx512M "
> SW_OAP_ADDRESS="http://sky-oap:12800"
> SW_TIMEOUT=20000
> ```

I think sky-oap in your SW_OAP_ADDRESS can not be resolved,you could use docker/docker-compose.yml file in skywalking project to start oap and ui.
